src/e2e/position.rs

```diff
@@ -0,0 +1,542 @@
+use crate::test_helpers::consts::GEAR_PATH;
+use crate::test_helpers::gclient::token::init_tokens;
+use crate::test_helpers::gclient::{
+    add_fee_tier, approve, balance_of, create_pool, create_position, get_api_user_id,
+    get_pool, get_position, get_tick, init_invariant, mint,
+};
+use contracts::{FeeTier, PoolKey};
+use decimal::*;
+use fungible_token_io::InitConfig;
+use gclient::{GearApi, Result};
+use gstd::prelude::*;
+use io::*;
+use math::{
+    fee_growth::FeeGrowth,
+    liquidity::Liquidity,
+    percentage::Percentage,
+    sqrt_price::{calculate_sqrt_price, SqrtPrice},
+};
+
+#[tokio::test]
+async fn test_create_position() -> Result<()> {
+    let admin_api = GearApi::dev_from_path(GEAR_PATH).await.unwrap();
+    let user_api = admin_api.clone().with("//Bob").unwrap();
+
+    let mut listener = admin_api.subscribe().await?;
+
+    assert!(listener.blocks_running().await?);
+
+    let init = InitInvariant {
+        config: InvariantConfig {
+            admin: get_api_user_id(&admin_api).into(),
+            protocol_fee: 100,
+        },
+    };
+
+    let invariant = init_invariant(&admin_api, &mut listener, init).await;
+
+    let (token_x, token_y) = init_tokens(&user_api, &mut listener, InitConfig::default()).await;
+
+    let init_tick = 0;
+    let init_sqrt_price = calculate_sqrt_price(init_tick).unwrap();
+
+    let fee_tier = FeeTier::new(Percentage::from_scale(2, 4), 1).unwrap();
+    add_fee_tier(&admin_api, &mut listener, invariant, fee_tier, None).await;
+
+    create_pool(
+        &user_api,
+        &mut listener,
+        invariant,
+        token_x,
+        token_y,
+        fee_tier,
+        init_sqrt_price,
+        init_tick,
+        None,
+    )
+    .await;
+    get_pool(&user_api, invariant, token_x, token_y, fee_tier, None)
+        .await
+        .unwrap();
+
+    mint(&user_api, &mut listener, token_x, 500).await.unwrap();
+    mint(&user_api, &mut listener, token_y, 500).await.unwrap();
+
+    approve(&user_api, &mut listener, token_x, invariant, 500)
+        .await
+        .unwrap();
+    approve(&user_api, &mut listener, token_y, invariant, 500)
+        .await
+        .unwrap();
+    let pool_key = PoolKey::new(token_x.into(), token_y.into(), fee_tier).unwrap();
+    create_position(
+        &user_api,
+        &mut listener,
+        invariant,
+        pool_key,
+        -10,
+        10,
+        Liquidity::new(10),
```

Comment:
We shouldn't abuse slippage in e2e tests unless it's specific test checking for slippage. It's best to pass the current price for both sides